contracts/sidechain/peripheral/SidechainClaimZap.sol

```diff
@@ -0,0 +1,245 @@
+        }
+
+        //claim from main curve LP pools
+        for (uint256 i = 0; i < rewardContracts.length; i++) {
+            IRewardStaking(rewardContracts[i]).getReward(msg.sender, true);
+        }
+        //claim from extra rewards
+        for (uint256 i = 0; i < extraRewardContracts.length; i++) {
+            IRewardStaking(extraRewardContracts[i]).getReward(msg.sender);
+        }
+        //claim from multi reward token contract
+        for (uint256 i = 0; i < tokenRewardContracts.length; i++) {
+            IRewardStaking(tokenRewardContracts[i]).getReward(msg.sender, tokenRewardTokens[i]);
+        }
+
+        // deposit/lock/stake
+        if (_callOptions(options)) {
+            _handleRewards(cvxBalance, cvxCrvBalance, amounts, options);
+        }
+
+        if (options.refundEth) {
+            (bool sent, bytes memory data) = payable(msg.sender).call{ value: address(this).balance }("");
+            require(sent, "!refund");
+        }
+    }
+
+    /**
+     * @notice returns a bool if handling of rewards should occur
+     * @param options                Claim options
+     */
+    function _callOptions(Options calldata options) internal view returns (bool) {
+        return (options.lockCvxL1 || options.sendCvxToL1 || options.useCompounder);
+    }
+
+    /**
+     * @notice  Bridge Rewards to L1 or Lock cvxCrv into compounders
+     * @param removeCvxBalance       cvxBalance to ignore and not redeposit (starting Cvx balance)
+     * @param removeCvxCrvBalance    cvxCrvBalance to ignore and not redeposit (starting CvxCrv balance)
+     * @param amounts                Claim rewards amoutns.
+     * @param options                see claimRewards
+     */
+    // prettier-ignore
+    function _handleRewards( // solhint-disable-line 
+        uint256 removeCvxBalance,   
+        uint256 removeCvxCrvBalance,       
+        ClaimRewardsAmounts calldata amounts, 
+        Options calldata options
+    ) internal {
+
+        address _l1receiver = options.overrideL1Receiver ? options.l1Receiever : msg.sender;
+
+        //Either lock cvx
+        if(options.lockCvxL1){
+            (uint256 cvxBalance, bool continued) = _checkBalanceAndPullToken(
+                cvx,
+                removeCvxBalance, 
+                amounts.lockCvxMaxAmount
+            );
+            if (continued) IAuraOFT(cvx).lock{value: msg.value}(cvxBalance, _l1receiver);
+        }
+        //or bridge it back to l1
+        else if (options.sendCvxToL1) {
+            (uint256 cvxBalance, bool continued) = _checkBalanceAndPullToken(
+                cvx,
+                removeCvxBalance, 
+                amounts.bridgeCvxMaxAmount
+            );
+
+            if (continued) IAuraOFT(cvx).sendFrom{value: msg.value}(
+                address(this), 
+                canonicalChainID, 
+                abi.encodePacked(_l1receiver), 
+                cvxBalance,
+                payable(_l1receiver), 
+                zro, 
```

Comment:
We probably want to make `zroPaymentAddress` and `adapterParams` a user input here. As we just need to pass them along to the `AuraOFT.sendFrom` function. So we can also remove the `zro` from storage and the setter.